datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +41,22 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
       List<String> unassignedPartitions) {
-    throw new NotImplementedException();
+    Validate.notNull(throughputInfo, "null throughputInfo");
+    Validate.notNull(assignedPartitions, "null assignedPartitions");
+    Validate.notNull(unassignedPartitions, "null unassignedPartitions");
+
+    Map<String, PartitionThroughputInfo> throughputMap = throughputInfo.getPartitionInfoMap();
+
+    Set<String> allPartitions = new HashSet<>(assignedPartitions);
+    allPartitions.addAll(unassignedPartitions);
+
+    // total throughput in KB/sec
+    int totalThroughput = throughputMap.entrySet().stream().
```

Comment:
1. Should you account some default weight for the partitions for which you did not get entry in throughput map? For eg. We may setup query to get all the partitions with more than 25KBps, in which case, we can assume the partitions with <25KBps to have some default weight rather than 0.

Reply:
That's an interesting point. Do you think that weight should be configurable?

Reply:
Yes, weight should be configurable. Please add that as well in this PR.